### PR TITLE
Reference management simplification

### PR DESCRIPTION
**Pull request by atstojanov**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

**Review by techsmyth — COMMENTED** [review]
Working very nicely on ecoverse. And code looks much cleaner.

However when I tried it on my user profile + adding multiple references I got errors...

Also saw the following error on the server:

image

File: src/components/Admin/Common/ContextReferenceSegment.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          input: {
+            ID: ref.id,
+          },
+        },
+      });
+      remove();
+    }
+  };
+
+  return <ReferenceSegment {...rest} onAdd={handleAdd} onRemove={handleRemove} readOnly={!contextId || readOnly} />;
```
Comment (hero101) [inline src/components/Admin/Common/ContextReferenceSegment.tsx:58]: Maybe a spread is not a good idea.

File: src/components/Admin/Common/ProfileReferenceSegment.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          input: {
+            ID: ref.id,
+          },
+        },
+      });
+      remove();
+    }
+  };
+
+  return <ReferenceSegment {...rest} onAdd={handleAdd} onRemove={handleRemove} readOnly={!profileId || readOnly} />;
```
Comment (hero101) [inline src/components/Admin/Common/ProfileReferenceSegment.tsx:59]: Maybe a spread is not a good idea.

File: src/components/Admin/Common/ReferenceSegment.tsx [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+import { useTranslation } from 'react-i18next';
 import * as yup from 'yup';
+import { Reference } from '../../../models/Profile';
 import Button from '../../core/Button';
 import Typography from '../../core/Typography';
-import { useTranslation } from 'react-i18next';
+import FormikInputField from './FormikInputField';
 
-interface ReferenceSegmentProps {
+export interface ReferenceSegmentProps {
```
Comment (hero101) [inline src/components/Admin/Common/ReferenceSegment.tsx:11]: Do you need that exported?

**Review by hero101 — CHANGES_REQUESTED** [review]

**Review by techsmyth — APPROVED** [review]
Working smoothly and fast for me :)

**Pull request merged**